### DISCUSSION TITLE
RawBlock html table: is it possible to use footnotes?

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#8061] [Q&A] [answered]
### Description

Hi,
I'm trying to use [HTML tables](But on this it is better to hear what John thinks.), and I have this [sample page](https://aborruso.github.io/linee-guida-opendata/small_table.html) (the [code](https://github.com/aborruso/linee-guida-opendata/blob/main/small_table.qmd)).

I have added a footnote using `data-qmd`, but it does not work, and I have the hard coded text in html output.

Am I doing something wrong?
Is it possible to use footnotes with RawBlock html tables?

Thank you

![image](https://github.com/quarto-dev/quarto-cli/assets/30607/5f7f23bd-2a6b-40cf-a55a-c3a11dba1e9f)

**Comment by cderv** [comment, marked as answer]
I do think this is a limitation of how we resolve the markdown passed into `data-qmd` attributes. I don't think the footnotes you put in the QMD document outside the table is seen. 

If you use the inline footnote syntax this will work

````markdown
---
format: html
keep-md: true
---

# Small table {.unnumbered}

```{=html}
<table>
    <tr>
        <th>Colonna 1</th>
        <th>Colonna 2</th>
        <th>Colonna 3</th>
    </tr>
    <tr>
        <td>Riga 1, Cellula 1 <span data-qmd="lorem ^[lorem ipsum]"></span></td>
        <td>Riga 1, Cellula 2</td>
        <td>Riga 1, Cellula 3</td>
    </tr>
    <tr>
        <td>Riga 2, Cellula 1</td>
        <td>Riga 2, Cellula 2</td>
        <td>Riga 2, Cellula 3</td>
    </tr>
</table>
```

````

![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/4b1120a3-dc0e-4cad-ad21-1e9d52e7a71c)

I'll open an issue in case this is something we can resolve (knowing the other markdown context to resolve the `data-qmd`

**Reply by aborruso** [reply]
Thank you @cderv, extremely useful. 

I have two things to add. The first one is long and I will add it later, it is related to my boring use case.

The second is related to cross reference and to the [special syntax](https://quarto.org/docs/books/book-crossrefs.html#creating-references) I can use to hide CR label and show only its number.

In [my example](https://aborruso.github.io/linee-guida-opendata/small_table.html) I have added [`<span data-qmd="[-@sec-soggetti-destinatari]"></span>`](https://github.com/aborruso/linee-guida-opendata/blob/main/small_table.qmd#L16), but in the output I have both the label and the ID.
Is it a bug or am I doing something wrong?

**Comment by cderv** [comment]
Side note regarding footnotes and tables: You could consider using some Table generation tools like **gt** that supports adding table footnotes quite precisely - see https://gt.rstudio.com/reference/tab_footnote.html

**Reply by aborruso** [reply]
I will try, but I'm not a R user and it seems to me very difficult 🙃
